src/diffusers/models/unet_3d_blocks.py

```diff
@@ -147,19 +160,30 @@ def __init__(
         attn_num_head_channels=1,
         output_scale_factor=1.0,
         cross_attention_dim=1280,
-        dual_cross_attention=False,
         use_linear_projection=True,
         upcast_attention=False,
+        use_temporal_transformer=True,  # False for TuneAVideo
+        use_temporal_conv=True,  # False for TuneAVideo
+        sub_blocks_type="2d",  # 2d for Text2VideoSD. #3d for TuneAVideo
     ):
         super().__init__()
 
         self.has_cross_attention = True
         self.attn_num_head_channels = attn_num_head_channels
         resnet_groups = resnet_groups if resnet_groups is not None else min(in_channels // 4, 32)
 
+        resnet_class = None
```

Comment:
It would be better if we could just create a new block here instead of having ot overwrite `resnet_class` and `transformer_class`. Ideally I'd like to avoid having to add a `sub_blocks_type` arg to the config of UNet3DConditionModel

Reply:
Also if we’re to overwrite the resnet and transformer classes without the ‘sub_blocks_type’ arg, how should I go about doing it ? Unet will need to know which blocks to use.

@patrickvonplaten

Reply:
@patrickvonplaten @sayakpaul need help on understanding this.

Even if we create the new blocks, how would they be passed to UNet3D config without an extra arg describing the block type ?

Reply:
Done Patrick. The new config looks like this: https://huggingface.co/NagaSaiAbhinay/tune-a-video-mo-di-bear-guitar-v1/blob/main/unet/config.json

Removes: sub_blocks_type, conv_type and replaces with custom blocks in up_block_types, down_block_types.
Adds: new mid_block_type
Unchanged: use_temporal_transformer

Doesn't affect the Text2Video pipelines.